src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1021,18 +1021,10 @@ bool PolicyHandler::SendMessageToSDK(const BinaryMessage& pt_string,
   LOG4CXX_AUTO_TRACE(logger_);
   POLICY_LIB_CHECK(false);
 
-  ApplicationSharedPtr app;
-  uint32_t app_id = 0;
-  if (last_used_app_ids_.empty()) {
-    LOG4CXX_WARN(logger_, "last_used_app_ids_ is empty");
-    return false;
-  } else {
-    app_id = last_used_app_ids_.back();
-
-    app = application_manager_.application(app_id);
-  }
+  uint32_t app_id = GetAppIdForSending();
+  ApplicationSharedPtr app = application_manager_.application(app_id);
 
-  if (!app) {
+  if (!app.valid()) {
```

Comment:
@LitvinenkoIra I think this change is not needed as operator `bool` is overloaded for SharedPtr and using valid() inside.